bitcoin/tests/psbt.rs

```diff
@@ -0,0 +1,446 @@
+    ];
+
+    // We pass the keys to the signer after doing verification to make explicit
+    // that signer is only using these two keys.
+    let keys = parse_and_verify_keys(&ext_priv, &test_vector);
+    let psbt_1 = signer_one_sign(psbt.clone(), keys);
+
+    //
+    // Step 5: The second signer.
+    //
+
+    // Strings from BIP 174 test vector.
+    let test_vector = vec![
+        ("cT7J9YpCwY3AVRFSjN6ukeEeWY6mhpbJPxRaDaP5QTdygQRxP9Au", "m/0h/0h/1h"),
+        ("cNBc3SWUip9PPm1GjRoLEJT6T41iNzCYtD7qro84FMnM5zEqeJsE", "m/0h/0h/3h"),
+    ];
+
+    let keys = parse_and_verify_keys(&ext_priv, &test_vector);
+    let psbt_2 = signer_two_sign(psbt, keys);
+
+    //
+    // Step 6: Combiner the two signed PSBTs.
+    //
+
+    let combined = combine(psbt_1, psbt_2);
+
+    //
+    // Step 7: Finalize the PSBT.
+    //
+
+    let finalized = finalize(combined);
+
+    //
+    // Step 8: Extract the transaction.
+    //
+
+    let _tx = extract_transaction(finalized);
+
+    //
+    // Step 9: Test lexicographical PSBT combiner.
+    //
+    // Combine would be done earlier, at Step 6, in typical workflow.
+    // We define it here to reflect the order of test vectors in BIP 174.
+    //
+
+    combine_lexicographically();
+}
+
+/// Attempts to build an extended private key from seed and also directly from a string.
+fn build_extended_private_key() -> ExtendedPrivKey {
+    // Strings from BIP 174 test vector.
+    let extended_private_key = "tprv8ZgxMBicQKsPd9TeAdPADNnSyH9SSUUbTVeFszDE23Ki6TBB5nCefAdHkK8Fm3qMQR6sHwA56zqRmKmxnHk37JkiFzvncDqoKmPWubu7hDF";
+    let seed = "cUkG8i1RFfWGWy5ziR11zJ5V4U4W3viSFCfyJmZnvQaUsd1xuF3T";
+
+    let xpriv = ExtendedPrivKey::from_str(extended_private_key).unwrap();
+
+    let sk = PrivateKey::from_wif(seed).unwrap();
+    let seeded = ExtendedPrivKey::new_master(NETWORK, &sk.inner.secret_bytes()).unwrap();
+    assert_eq!(xpriv, seeded);
+
+    xpriv
+}
+
+/// Creates the initial transaction, called by the PSBT Creator.
+fn create_transaction() -> Transaction {
+    // Strings from BIP 174 test vector.
+    let output_0 = TvOutput {
+        amount: "1.49990000",
+        script_pubkey: "0014d85c2b71d0060b09c9886aeb815e50991dda124d",
+
+    };
+    let output_1 = TvOutput {
+        amount: "1.00000000",
+        script_pubkey: "001400aea9a2e5f0f876a588df5546e8742d1d87008f",
+    };
+    let input_0 = TvInput {
+        txid: "75ddabb27b8845f5247975c8a5ba7c6f336c4570708ebe230caf6db5217ae858",
+        index: 0,
+    };
+    let input_1 = TvInput {
+        txid: "1dea7cd05979072a3578cab271c02244ea8a090bbb46aa680a65ecd027048d83",
+        index: 1,
+    };
+    struct TvOutput {
+        amount: &'static str,
+        script_pubkey: &'static str,
+    }
+    struct TvInput {
+        txid: &'static str,
+        index: u32,
+    }
+
+    Transaction {
+        version: 2,
+        lock_time: bitcoin::absolute::PackedLockTime::ZERO,
```

Comment:
nit: With the addition of `use bitcoin::locktime::absolute;

```suggestion
        lock_time: absolute::PackedLockTime::ZERO,
```